sys/include/net/ng_fib/ng_fib.h

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (C) 2014  Martin Landsmann <Martin.Landsmann@HAW-Hamburg.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_fib Forwarding Information Base (FIB)
+ * @ingroup     net
+ * @brief       FIB implementation
+ *
+ * @{
+ *
+ * @file
+ * @brief       Types and functions for FIB
+ * @author      Martin Landsmann
+ */
+
+#ifndef FIB_H_
+#define FIB_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Reactive Routing Protocol (RRP) message content to request/reply route discovery
+ */
+typedef struct rrp_address_msg_t {
+    uint8_t *address;      /**< The pointer to the address */
+    uint8_t address_size;  /**< The address size */
+    uint32_t address_flags; /**< The flags for the given address */
+    kernel_pid_t iface_id; /**< The interface used for the next-hop reply.
+                                If it is KERNEL_PID_UNDEF, no next-hop has been found. */
+    uint32_t lifetime;     /**< The lifetime of this next-hop */
+} rrp_address_msg_t;
+
+#define FIB_MSG_RRP_SIGNAL (0x99)     /**< message type for RRP notifications         */
+
+#define FIB_SUCCESS_EXACT_MATCH (1)   /**< return value for successfull exact match   */
+#define FIB_SUCCESS (0)               /**< return value for success                   */
```

Comment:
for the RRP? If yes, we still need to tell the RRP which destination we are looking for, but we can get rid of the lifetime and iface_id in the message